be/src/olap/task/engine_publish_version_task.cpp

```diff
@@ -189,8 +189,11 @@ Status EnginePublishVersionTask::finish() {
                 }
             }
             auto rowset_meta_ptr = rowset->rowset_meta();
-            _tablet_id_to_num_delta_rows->insert(
-                    {rowset_meta_ptr->tablet_id(), rowset_meta_ptr->num_rows()});
+            auto table_id = StorageEngine::instance()
+                                    ->tablet_manager()
+                                    ->get_tablet(rowset_meta_ptr->tablet_id())
+                                    ->get_table_id();
+            (*_table_id_to_num_delta_rows)[table_id] += rowset_meta_ptr->num_rows();
```

Comment:
You should not accumulate the total rows here, publish tasks might fail and retry, you'd better to move such code to L232

Reply:
Modified.